integrations/nifi/nifi-nar-bundles/nifi-hadoop-bundle/nifi-hadoop-processors/src/main/java/com/thinkbiganalytics/nifi/v2/hdfs/AbstractHadoopProcessor.java

```diff
@@ -199,6 +199,19 @@ public final void abstractOnScheduled(ProcessContext context) throws IOException
         }
     }
 
+    protected void resetFileSystem() throws IOException {
+        HdfsResources resources = hdfsResources.get();
+        Configuration conf = resources.configuration;
+        UserGroupInformation ugi = resources.userGroupInformation;
+        FileSystem fs = resources.fileSystem;
+        if (fs != null) {
+            fs.close();
```

Comment:
This will be called each time the processor is scheduled. For the default Kylo flow, the processor is set to run every 30 seconds. The filesystem would be closed and opened every 30 seconds, which is quite expensive operation. A more efficient approach is desirable.